test/e2e/utils/utils.go

```diff
@@ -52,9 +55,16 @@ func GetPodLogs(ctx context.Context, f *framework.Framework, podName, containerN
 	return fmt.Sprintf("log output of the container %s in the pod %s:%s", containerName, podName, log)
 }
 
-// WaitForNodesWithResource waits for nodes to have positive allocatable resource.
-func WaitForNodesWithResource(ctx context.Context, c clientset.Interface, res v1.ResourceName, timeout time.Duration) error {
-	framework.Logf("Waiting up to %s for any positive allocatable resource %q", timeout, res)
+// WaitForNodesWithResource waits for node's resources to change.
+// Depending on the waitOperation, function waits for positive resource count or a zero resource count.
+func WaitForNodesWithResource(ctx context.Context, c clientset.Interface, res v1.ResourceName, timeout time.Duration, waitOperation int) error {
```

Comment:
This function does feel a bit cumbersome. The operation being passed as an int requires the if-else soup, plus the name of WaitOpGreater is a bit confusing in a sense since it really means waiting for resource amount to be greater than zero.

Now this is just a test so not really a big deal, but the typical way of doing these is to pass in a function which performs the logical operation, and that can then be written on the fly (but doesn't have to). But then doing the logging, if you need it, is not possible (because you might not want to log inside two for-loops).

If you want to keep this the way it is (because of logging etc) at least rename the WaitOpGreater to something like WaitOpGreaterThanZero.

Reply:
yeah, the function is a bit clumsy. I'll have another stab at it.